src/dot-protocol/merlin_dot_protocol.ml

```diff
@@ -127,39 +127,74 @@ module Sexp = struct
     List (List.map ~f directives)
 end
 
-module Csexp = Csexp.Make (Sexp)
+type read_error =
+  | Unexpected_output of string
+  | Csexp_parse_error of string
 
-module Commands = struct
-  type t = File of string | Halt | Unknown
+module Make (IO : sig
+    type 'a t
+
+    module O : sig
+      val ( let+ ) : 'a t -> ('a -> 'b) -> 'b t
+    end
+  end) (Chan : sig
```

Comment:
Indentation of the arguments feels off. I would endline before each.
(hopefully we will soon ocamlformat the codebase and don't do this kind of comments anymore)

Reply:
Done, I've used ocamlformat with config from #1586 . I hope it's fine now :)